plugins/query-insights/src/main/java/org/opensearch/plugin/insights/core/service/TopQueriesService.java

```diff
@@ -169,6 +203,40 @@ public void validateWindowSize(final TimeValue windowSize) {
         }
     }
 
+    /**
+     * Set up the top queries exporter based on provided settings
+     *
+     * @param settings exporter config {@link Settings}
+     */
+    public void setExporter(final Settings settings) {
+        if (settings.get(EXPORTER_TYPE) != null) {
+            SinkType expectedType = SinkType.parse(settings.get(EXPORTER_TYPE, DEFAULT_TOP_QUERIES_EXPORTER_TYPE));
+            if (exporter != null && expectedType == SinkType.getSinkTypeFromExporter(exporter)) {
+                queryInsightsExporterFactory.updateExporter(
+                    exporter,
+                    settings.get(EXPORT_INDEX, DEFAULT_TOP_N_LATENCY_QUERIES_INDEX_PATTERN)
+                );
```

Comment:
Toggling the setting between debug/local_index sink types can create new objects which won't be garbage collected due to reference from the list in exporter.

Reply:
good point. Let me update the related logic